backend-lucene/src/main/java/org/hibernate/search/v6poc/backend/lucene/search/predicate/impl/BooleanJunctionPredicateBuilderImpl.java

```diff
@@ -56,6 +97,7 @@ private void mustNot(Query luceneQuery) {
 
 	private void should(Query luceneQuery) {
 		booleanQueryBuilder.add( luceneQuery, Occur.SHOULD );
+		++shouldClauseCount;
```

Comment:
First time I see the increment syntax in this order when you don't explicitly want this behavior.